_posts/2024-02-01-wildfly-glow.adoc

```diff
@@ -0,0 +1,504 @@
+
+[source,bash]
+----
+wildfly-glow.sh scan myapp.war --add-ons=postgresql --provision=SERVER
+----
+Note: the provisioned server contains the deployment that has been scanned.
+
+Note: by specifying `--provision=BOOTABLE_JAR` you can produce a WildFly bootable JAR. By specifying `--provision=DOCKER_IMAGE` you can produce a Docker image.
+
+Producing a Docker image for cloud execution example:
+
+[source,bash]
+----
+wildfly-glow.sh scan myapp.war --add-ons=postgresql --provision=DOCKER_IMAGE --cloud
+----
+
+===== Output example
+
+WildFly Glow CLI output what it has discovered, for example:
+
+[source,bash]
+----
+
+wildfly-glow.sh scan myapp.war 
+
+Wildfly Glow is scanning...
+context: bare-metal
+enabled profile: none
+galleon discovery
+- feature-packs
+   org.wildfly:wildfly-galleon-pack:30.0.1.Final
+- layers
+   ee-core-profile-server
+   jpa
+   ejb-lite
+   jaxrs
+   jsf
+   h2-driver
+
+Some suggestions have been found. You could enable suggestions with --suggest option (if using the WildFly Glow CLI) or <suggest>true</suggest> (if using the WildFly Maven Plugin).
+To provision the WildFly server for your deployment add the --provision=SERVER option to the scan command
+
+----
+
+== Understanding why a Galleon layer has been selected
+
+WildFly Glow is based on rules contained in the Wildfy Galleon layers. If a rule matches, the layer is included. The set of rules 
+contained in WildFly layers is documented in this http://docs.wildfly.org/wildfly-galleon-feature-packs/[documentation].
+
+When enabling verbose mode (`--verbose` option with WildFly Glow CLI, `<verbose>true</verbose>` option in the maven plugin), 
+WildFly Glow will output for each layer the rules that matched.
+
+An example of output:
+
+----
+...
+layers inclusion rules
+* ee-core-profile-server
+  - BASE_LAYER
+* ee-concurrency
+  - JAVA_TYPE: [jakarta.enterprise.concurrent.*]
+* undertow-https
+  - ADD_ON
+...
+----
+
+This output means:
+
+* `ee-core-profile-server` is a base layer (always included).
+* `ee-concurrency` layer is included because a Java class located in the java package `jakarta.enterprise.concurrent` has been found.
+* `undertow-https` is included because it is bound to an included add-on (in this case `ssl`).
+
+== Trimming numbers with WildFly Glow
+
+We have measured the Disk usage and Memory consumption of some WildFly quickstarts 
+(using XXXTODOLINK WildFly Glow branch vs not using XXXTODOLINK WildFly Glow branch). We have observed a reduction of 5% to 55% for disk usage 
+and 5% to 32% for memory consumption. Variation bound to the complexity of the quickstart. If the quickstart requires all the Galleon layers 
+present in the aggegator layer (eg: `cloud-server` or `jaxrs-server`) then the gain is lower.
+
+My colleague Kabir Khan has wrote an interesting https://github.com/kabir/vlog-glow/blob/main/README.md[project] 
+and associated Vlog that showcases the gain you can expect with WildFly Glow. If you are interested by WildFly Glow 
```

Comment:
It isn't my Vlog :-) They should follow the [WIldFlyAs channel](https://www.youtube.com/@WildFlyAS)  but that is of course more general WildFly stuff.

They can of course watch my specific video.